lms/djangoapps/onboarding_survey/middleware.py

```diff
@@ -22,12 +23,15 @@ class RedirectMiddleware(object):
 
     def process_request(self, request):
 
-        if request.is_ajax():
+        if request.is_ajax() or request.get_full_path() == '/logout':
             return None
 
         if not request.user.is_anonymous():
             user = request.user
 
+            if user.username == settings.ADMIN_USER_NAME:
```

Comment:
what if we have multiple admin users? please get admin info from user object. user.is_superuser